test/e2e/e2e.go

```diff
@@ -713,33 +690,492 @@ var _ = Describe("validation", func() {
 						},
 					},
 				}
+				err := testclient.Client.Delete(context.Background(), addresspool)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Checking ConfigMap is deleted")
 				Eventually(func() bool {
-					err := testclient.Client.Delete(context.Background(), addresspool)
+					_, err := testclient.Client.ConfigMaps(OperatorNameSpace).Get(context.Background(), consts.MetalLBConfigMapName, metav1.GetOptions{})
 					return errors.IsNotFound(err)
-				}, timeout, interval).Should(BeTrue(), "Failed to delete AddressPool custom resource")
+				}, timeout, interval).Should(BeTrue())
 
-				By("Checking ConfigMap is deleted")
+			})
+
+			// Make sure Configmap is deleted at the end of this test
+			By("Checking ConfigMap is deleted at the end of the test")
+			Eventually(func() bool {
```

Comment:
this is done twice. maybe we can update the test to check once that the configmap was deleted.